src/components/Input/Input.tsx

```diff
@@ -14,37 +14,35 @@ export interface InputProps {
 
   /** Whether the input should be disabled. */
   readonly isDisabled?: boolean;
+
+  /** The element that will be placed on the left side of the input. */
+  readonly leftElement?: ReactNode;
 }
 
 interface Props extends InputProps {
 
-  /** Name attribute. */
-  readonly name: string;
-
-  /** Input value. */
-  readonly value: string;
-
-  /** Input change callback. */
-  readonly onChange: ChangeEventHandler<HTMLInputElement>;
-
-  /** Type attribute. Default value is `text`. */
-  readonly type?: HTMLInputTypeAttribute;
-
   /** The element that will be placed on the right side of the input. */
   readonly rightElement?: ReactNode;
 }
 
-/** Input. */
-const InputComponent = forwardRef<HTMLInputElement, Props>(({
+/**
+ * Input.
+ * @param ref Forwarded ref.
+ */
+const InputComponent: FC<Props> = ({
   id,
   name,
   value,
   onChange,
   placeholder,
   isDisabled,
   type,
+  leftElement,
+  onBlur,
   rightElement,
-}, ref) => {
+  onFocus,
+  className,
+}, ref: ForwardedRef<HTMLInputElement>) => {
```

Comment:
Since you extended props type, make sure that other attributes are also applied to the input. You can do it by using the spread notation `...props`